miio/airfresh.py

```diff
@@ -0,0 +1,292 @@
+import enum
+import logging
+from collections import defaultdict
+from typing import Any, Dict, Optional
+
+import click
+
+from .click_common import command, format_output, EnumType
+from .device import Device, DeviceException
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AirFreshException(DeviceException):
+    pass
+
+
+class OperationMode(enum.Enum):
+    # Supported modes of the Air Fresh VA2
+    Auto = 'auto'
+    Silent = 'silent'
+    Interval = 'interval'
+    Low = 'low'
+    Middle = 'middle'
+    Strong = 'strong'
+
+
+class LedBrightness(enum.Enum):
+    Bright = 0
+    Dim = 1
+    Off = 2
+
+
+class AirFreshStatus:
+    """Container for status reports from the air fresh."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self.data = data
+
+    @property
+    def power(self) -> str:
+        """Power state."""
+        return self.data["power"]
+
+    @property
+    def is_on(self) -> bool:
+        """Return True if device is on."""
+        return self.power == "on"
+
+    @property
+    def aqi(self) -> int:
+        """Air quality index."""
+        return self.data["aqi"]
+
+    @property
+    def average_aqi(self) -> int:
+        """Average of the air quality index."""
+        return self.data["average_aqi"]
+
+    @property
+    def co2(self) -> int:
+        """Carbon dioxide."""
+        return self.data["co2"]
+
+    @property
+    def humidity(self) -> int:
+        """Current humidity."""
+        return self.data["humidity"]
+
+    @property
+    def temperature(self) -> Optional[float]:
+        """Current temperature, if available."""
+        if self.data["temp_dec"] is not None:
+            return self.data["temp_dec"] / 10.0
+
+        return None
+
+    @property
+    def mode(self) -> OperationMode:
+        """Current operation mode."""
+        return OperationMode(self.data["mode"])
+
+    @property
+    def led_brightness(self) -> Optional[LedBrightness]:
+        """Brightness of the LED."""
+        if self.data["led_level"] is not None:
+            try:
+                return LedBrightness(self.data["led_level"])
+            except ValueError:
+                return None
+
+        return None
+
+    @property
+    def buzzer(self) -> Optional[bool]:
+        """Return True if buzzer is on."""
+        if self.data["buzzer"] is not None:
+            return self.data["buzzer"] == "on"
+
+        return None
+
+    @property
+    def child_lock(self) -> bool:
+        """Return True if child lock is on."""
+        return self.data["child_lock"] == "on"
```

Comment:
Should this and the other properties also be protected against None?

Reply:
This property is always available and doesn't need to be safeguarded.